### PR TITLE
Dmirano/4238 cypress budget checks

### LINKED ISSUE #4238: [Cypress] add budget checks into APD tests or bug tests
Jerome would like us to add budget checks to the APD tests that are checked as you are putting in data to make sure it's being added to the correct places of the budget. These tests should be added to the apd-data tests.

### Test Cases

| Given | When | Then |
|---|---|---|
| An APD | add a key personnel to activity 1 that doesn't have direct costs | the budget should not be affected |
| An APD | add a key personnel to activity 1 that does have direct costs | the budget should be updated to show those costs in HIT state staff and in-house costs |
| An APD | add a state personnel to an HIT activity | the budget should be updated to show those costs in HIT state staff and in-house costs |
| An APD | add a state expense to an HIT activity | the budget should be updated to show those costs in HIT state expenses and in-house costs |
| An APD | add a contractor to an HIT activity | the budget should be updated to show those costs in HIT contractors costs |
| An APD | add a state personnel to an HIE activity | the budget should be updated to show those costs in HIE state staff and in-house costs |
| An APD | add a state expense to an HIE activity | the budget should be updated to show those costs in HIE state expenses and in-house costs |
| An APD | add a contractor to an HIE activity | the budget should be updated to show those costs in HIE contractors costs |
| An APD | add a state personnel to an MMIS activity | the budget should be updated to show those costs in MMIS state staff and in-house costs |
| An APD | add a state expense to an MMIS activity | the budget should be updated to show those costs in MMIS state expenses and in-house costs |
| An APD | add a contractor to an MMIS activity | the budget should be updated to show those costs in MMIS contractors costs |
| An APD | add a state personnel to an activity with no funding source | the budget should be updated to show those costs in the total values but not in the HIT/HIE/MMIS tables |
| An APD | add a state expense to an activity with no funding source | the budget should be updated to show those costs in the total values but not in the HIT/HIE/MMIS tables |
| An APD | add a contractor to an activity with no funding source | the budget should be updated to show those costs in the total values but not in the HIT/HIE/MMIS tables |

### This task is done when...
- [ ] all of the test cases have been added

### PR DESCRIPTION
**Pull request by mirano-darren**
Resolves #4238 

### Description
Adds Cypress tests to check the budget calculations.

### Significant changes or possible side effects
Adds the test suite to the circle config

### Automated test cases written
04-budget-checks

### Steps to manually verify this change

1. Verify test runs and passes in circle

### This pull request is ready to review when

- [ ] Automated tests are updated (and all tests are passing)
- [ ] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor
- [ ] Associated OpenAPI documentation has been updated
- [ ] The experience passes a basic manual accessibility audit (keyboard nav,
      screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the functionality related to the change
- [ ] QA has verified the accessibility using tools such as screen readers and
      WAVE (accessibility evaluation tool)
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -559,6 +559,17 @@ jobs:
             if [[ $CIRCLE_NODE_INDEX == "3" ]]; then export DEV_DB_PORT=54325; fi
             if [[ $CIRCLE_NODE_INDEX == "3" ]]; then export CYPRESS_RECORD_KEY="$CYPRESS_RECORD_KEY"; fi
             if [[ $CIRCLE_NODE_INDEX == "3" ]]; then yarn run cy:run:e2e-ci -- --record --key $CYPRESS_RECORD_KEY --ci-build-id "$CIRCLE_BUILD_NUM-$CIRCLE_BRANCH" --group "e2e tests apd data" --spec "cypress/integration/02-apd/03-apd-with-data.cy.js"; fi
+      - run:
```
Comment (thetif) [inline .circleci/config.yml:562]: the index can't be the same, you'll have to change this to 4 and the next one to 5 and the next one to 7

Comment (thetif) [reply .circleci/config.yml:562]: you also need to update the parallelism flag up above to 7

File: e2e/cypress/integration/02-apd/04-apd-budget-checks.cy.js [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+    cy.fixture('budget-checks-test.json').then(data => {
+      budgetData = data;
+    });
+  });
+
+  beforeEach(() => {
+    cy.visit(apdUrl);
+  });
+
+  // after(() => {
```
Comment (thetif) [inline e2e/cypress/integration/02-apd/04-apd-budget-checks.cy.js:48]: don't you want to delete this in the end

File: e2e/cypress/page-objects/fill-out-activity-page.js [hunk 3]
```diff
@@ -73,30 +73,59 @@ class FillOutActivityPage {
...
-        staff.ftes
+        staffList[i].title,
+        staffList[i].description,
+        staffList[i].costs,
+        staffList[i].ftes
       );
-    });
+    }
+
+    // _.forEach(staffList, (staff, i) => {
```
Comment (thetif) [inline e2e/cypress/page-objects/fill-out-activity-page.js:109]: did the forEach version not work? can you remove this commented out code

File: e2e/cypress/page-objects/fill-out-activity-page.js [hunk 4]
```diff
@@ -124,28 +154,55 @@ class FillOutActivityPage {
...
-        expense.costs,
-        expense.description
+        expenseList[i].category,
+        expenseList[i].costs,
+        expenseList[i].description
       );
-    });
+    }
+
+    // _.forEach(expenseList, (expense, i) => {
```
Comment (thetif) [inline e2e/cypress/page-objects/fill-out-activity-page.js:188]: see note above

File: e2e/cypress/page-objects/fill-out-activity-page.js [hunk 5]
```diff
@@ -174,19 +232,36 @@ class FillOutActivityPage {
...
+    testDelete = false,
+    testBudget = false
+  ) => {
     cy.findByRole('heading', {
       name: /Private Contractor Costs/i,
       level: 3
     }).should('exist');
 
-    _.forEach(contractorList, (contractor, i) => {
+    // _.forEach(contractorList, (contractor, i) => {
```
Comment (thetif) [inline e2e/cypress/page-objects/fill-out-activity-page.js:246]: remove commented code

**Review by thetif — CHANGES_REQUESTED** [review]

**Review by tbolt — APPROVED** [review]
Good once @thetif's feedback is addressed 👍

File: e2e/cypress/page-objects/fill-out-activity-page.js (hunk 3 above)
Comment (mirano-darren) [reply e2e/cypress/page-objects/fill-out-activity-page.js:109]: I couldn't use `break `to get out of a forEach() so I changed them to for loops

Comment (thetif) [reply e2e/cypress/page-objects/fill-out-activity-page.js:109]: ahhh could you make a comment saying that above the tests just so no one comes in and tries to make it a forEach again, and then remove the commented code.

**Review by thetif — APPROVED** [review]

**Pull request merged**